accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,790 @@
+  be possible to produce a single binary that can use newer APIs when calls are
+  properly guarded with an OS check.
+
+### Non-Goals
+
+* Replace TFMs or expand runtime identifiers (RIDs)
+
+## Design
+
+We'll have the following TFMs:
+
+| TFM             | Compatible With                                            | Comments                          |
+|-----------------|------------------------------------------------------------|-----------------------------------|
+| net5.0          | net1..4 (with NU1701 warning)                              | No WinForms or WPF                |
+|                 | netcoreapp1..3.1 (warning when WinForms/WPF is referenced) |                                   |
+|                 | netstandard1..2.1                                          |                                   |
+| net5.0-android  | xamarin.android                                            |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-ios      | xamarin.ios                                                |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-macos    | xamarin.mac                                                |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-tvos     | xamarin.tvos                                               |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-watchos  | xamarin.watchos                                            |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-windows  | netcoreapp1..3.1                                           | WinForms + WPF                    |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| Tizen, Unity... | Will follow the Xamarin model                              |                                   |
+
+_**Open Issue**. In case of Xamarin, there is a discussion of whether the
+convergence with .NET Core will be a breaking change for existing NuGet packages
+(for example, there is a desire to change their namespaces). In that case, we
+should not map it to the existing TFM as that would be pointless._
+
+_**Open Issue**. Will the above scheme work for iPad?_
+
+### OS versions
+
+The TFM will support an optional OS version, such as:
+
+* `net5.0-ios`
+* `net5.0-ios12.0`
+
+A TFM without an OS version will be interpreted as the lowest OS version that
+was supported by the corresponding `netX.Y` version.
```

Comment:
Based on the text after the note, that would be **highest** (not lowest)

Why ? as of today Xamarin.iOS supports iOS 7.0 (and later) up to iOS 13.3.

We do not want `net-5.0-ios` to default to iOS 7.0 (the lowest).

The current behaviour is to default to the **highest** (13.3) and have the developer guard newer API calls (when needed) to ensure the code works on all versions of iOS they wish to support.

Reply:
That also means a lot of `net5.0-ios-*.*` TFM - about 29 today.

https://github.com/xamarin/xamarin-macios/blob/master/Versions-ios.plist.in#L11

Reply:
> Based on the text after the note, that would be **highest** (not lowest)
> 
> Why ? as of today Xamarin.iOS supports iOS 7.0 (and later) up to iOS 13.3.

I think lowest might not be the best way to think about. It's the *first* version that was bundled with a given version of .NET. For example, say .NET 5 will initially bundle iOS 13 and with a later updated iOS 14. .NET 6 might bundle iOS 14 and then in a later version iOS 16. The spec prescribes the meaning as follows:

* `net5-ios` is equivalent to `net5-ios13`
* `net6-ios` is equivalent to `net5-ios14`

Does that make more sense?

Reply:
yup, it's what I kind of expected 👍